crates/prover/src/components/opcodes/u32_store_add_fp_imm.rs

```diff
@@ -154,53 +175,128 @@ impl Claim {
-                *row[12] = dst_prev_clock;
-                *row[13] = dst_prev_val;
+                *row[5] = src_off;
+                *row[6] = imm_lo;
+                *row[7] = imm_hi;
+                *row[8] = dst_off;
+                *row[9] = op0_val_lo;
+                *row[10] = op0_val_hi;
+                *row[11] = op0_prev_clock;
+                *row[12] = dst_prev_val_lo;
+                *row[13] = dst_prev_val_hi;
+                *row[14] = dst_prev_clock;
+                *row[15] = u16_carry;
+                *row[16] = u32_carry;
 
                 *lookup_data.registers[0] = [input.pc, input.fp];
-                *lookup_data.registers[1] = [input.pc + one, input.fp];
-
-                *lookup_data.memory[0] =
-                    [input.pc, inst_prev_clock, opcode_constant, off0, off1, off2];
-                *lookup_data.memory[1] = [input.pc, clock, opcode_constant, off0, off1, off2];
-
-                *lookup_data.memory[2] = [fp + off0, op0_prev_clock, op0_val, zero, zero, zero];
-                *lookup_data.memory[3] = [fp + off0, clock, op0_val, zero, zero, zero];
-
-                *lookup_data.memory[4] = [fp + off1, op1_prev_clock, op1_val, zero, zero, zero];
-                *lookup_data.memory[5] = [fp + off1, clock, op1_val, zero, zero, zero];
-
-                *lookup_data.memory[6] =
-                    [fp + off2, dst_prev_clock, dst_prev_val, zero, zero, zero];
-                *lookup_data.memory[7] = [fp + off2, clock, op0_val - op1_val, zero, zero, zero];
+                *lookup_data.registers[1] = [input.pc + one + one, input.fp];
+
+                // Read first QM31 word for instruction
+                *lookup_data.memory[0] = [
+                    input.pc,
+                    inst_prev_clock,
+                    opcode_constant,
+                    src_off,
+                    imm_lo,
+                    imm_hi,
+                ];
+                *lookup_data.memory[1] =
+                    [input.pc, clock, opcode_constant, src_off, imm_lo, imm_hi];
+
+                // Read second QM31 word for instruction
+                *lookup_data.memory[2] =
+                    [input.pc + one, inst_prev_clock, dst_off, zero, zero, zero];
+                *lookup_data.memory[3] = [input.pc + one, clock, dst_off, zero, zero, zero];
+
+                // Read op0_lo
+                *lookup_data.memory[4] =
+                    [fp + src_off, op0_prev_clock, op0_val_lo, zero, zero, zero];
+                *lookup_data.memory[5] = [fp + src_off, clock, op0_val_lo, zero, zero, zero];
+
+                // Read op0_hi
+                *lookup_data.memory[6] = [
+                    fp + src_off + one,
+                    op0_prev_clock,
+                    op0_val_hi,
+                    zero,
+                    zero,
+                    zero,
+                ];
+                *lookup_data.memory[7] = [fp + src_off + one, clock, op0_val_hi, zero, zero, zero];
```

Comment:
iiuc, this is a mistake as `hi` is just the second limb at address `src_off` and not the first limb at the next address, since each address stores a QM31